File: src/main/java/com/example/doorcenter/frame/Frame.java
package com.example.doorcenter.frame;

import com.example.doorcenter.platband.PlatbandVariant;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@AllArgsConstructor
@Setter
@NoArgsConstructor
public class Frame {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;
    private int width;
    private int thickness;
    private String type;
    private String imageUrl;
    private String color;

    @OneToMany(mappedBy = "frame", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<FrameVariant> variants; // List of height-width-price combinations

}

File: src/app/models/frame.ts
import { PlatbandVariant } from "./platband-variant";
import { ProductMedia } from "./product-media";

export interface Frame {
    id: number;
    width: number;
    thickness: number;
    type: string;
    color: string;
    imageUrl: string;
    videoUrl?: string; // Optional video URL
    variants: PlatbandVariant[] // Array of variants for height, width, and price
    media?: ProductMedia[]; // Array of media files (images and videos)
}